pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1191,7 +1213,25 @@ func (vs *VSphere) NodeDeleted(obj interface{}) {
 	}
 
 	glog.V(4).Infof("Node deleted: %+v", node)
-	vs.nodeManager.UnRegisterNode(node)
+	go func() {
+		nodeName := convertToK8sType(node.ObjectMeta.Name)
+		for i := 0; i < CheckDisksOnNodeDeletedMaxRetry; i++ {
+			disks := vs.nodeManager.GetAttachedDisks(nodeName)
```

Comment:
I think if controller manager is restarted in somewhere at this point then detach disk will still fail because detach disk won't be able to find the *v1.Node for the respective node.

Reply:
Thanks for reviewing. I think we could manage that case by keeping removed-node information as I did in 1st commit in addition to attached-disks info, until we rewrite the code using vm uuid as nodename. However this could happen when 3 unusual things happen at the same time: (1) node shutdown on which some containers running (2) problem connecting to vcenter (3) controller mangers restarted in other nodes.
I'm wondering if we could go with current way instead of adding some complexity to solve it for now.

Reply:
I agree with Abrar. Even though the 3 things you mentioned may not happen at the same time, this is not a reliable solution. Instead of pushing this change now, why not have a complete solution? 

Any reason why you want to push this change soon?